lib/modules/accountSwitcher.js

```diff
@@ -434,7 +459,37 @@ function _notifySwitchedAccountElsewhere() {
 		message,
 	});
 }
-
+function autoSwitcher(thing) {
+	const loggedInUsername = loggedInUser();
+	if (!loggedInUsername) return;
+	const currentUserIsAuthor = thing.getAuthor() === loggedInUsername;
+	if (!currentUserIsAuthor && !isAutoSwitched) {
+		sessionStorage.setItem('RESAccountSwitcherAutoSwitched', thing.isPost() ? 'post' : 'comment');
+		switchTo(thing.getAuthor());
+	}
+	if (isAutoSwitched === 'post' && currentUserIsAuthor && !autoSwitchNotificationShown) {
+		Notifications.showNotification({
+			moduleID: module.moduleID,
+			notificationID: 'auto-switched',
+			optionKey: 'autoSwitchOnPost',
+			header: i18n('accountSwitcherAutoSwitchNotificationTitle'),
+			message: i18n('accountSwitcherAutoSwitchPostDesc', loggedInUsername),
+		});
+		autoSwitchNotificationShown = true;
```

Comment:
The notifications ought to be separated out to a new `_.once` function. That way it can only be invoked once, and `autoSwitchNotificationShown` becomes redundant.